pkg/backup/restore/restore_manager.go

```diff
@@ -66,21 +66,30 @@ func (rm *restoreManager) syncRestoreJob(restore *v1alpha1.Restore) error {
 		var tc *v1alpha1.TidbCluster
 		tc, err = rm.deps.TiDBClusterLister.TidbClusters(restoreNamespace).Get(restore.Spec.BR.Cluster)
 		if err != nil {
-			klog.Errorf("failed to fetch tidbcluster %s/%s, error: %s", restoreNamespace, restore.Spec.BR.Cluster, err.Error())
-			return nil
+			reason := fmt.Sprintf("failed to fetch tidbcluster %s/%s", restoreNamespace, restore.Spec.BR.Cluster)
+			rm.statusUpdater.Update(restore, &v1alpha1.RestoreCondition{
+				Type:    v1alpha1.RestoreRetryFailed,
+				Status:  corev1.ConditionTrue,
+				Reason:  reason,
+				Message: err.Error(),
+			})
+			return err
 		}
 
 		tikvImage := tc.TiKVImage()
 		err = backuputil.ValidateRestore(restore, tikvImage)
 	}
 
 	if err != nil {
-		rm.statusUpdater.Update(restore, &v1alpha1.RestoreCondition{
+		updateErr := rm.statusUpdater.Update(restore, &v1alpha1.RestoreCondition{
 			Type:    v1alpha1.RestoreInvalid,
 			Status:  corev1.ConditionTrue,
 			Reason:  "InvalidSpec",
 			Message: err.Error(),
 		})
+		if updateErr != nil {
+			klog.Warning(updateErr)
+		}
 
 		return controller.IgnoreErrorf("invalid restore spec %s/%s", ns, name)
```

Comment:
Suggest that the change here should be consistent with that in the backup_manager.go.

Reply:
revert this change: https://github.com/pingcap/tidb-operator/pull/3446/commits/674af126109791f567771b37785f69a2c29e1f97
There many other `Updata` in restore_manager.go too, better not just ignore the error.
We can handle it in a consistent way later.